### PR TITLE
fix-6212-Issue-With-Linkedin-OAuth

### PR DESCRIPTION
**Pull request by btme0011**
## What does this PR do?
Fix related to the issue - #6212 

Reference document - https://learn.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/sign-in-with-linkedin-v2?context=linkedin%2Fconsumer%2Fcontext

LinkedIn has transitioned to using OpenID for authentication and authorization, necessitating updates to the existing codebase to accommodate these changes.
Screenshot 2023-09-13 at 9 42 03 PM

## Test Plan

Screenshot 2023-09-14 at 1 18 31 AM

## Related PRs and Issues

- #6212 

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

File: src/Appwrite/Auth/OAuth2/Linkedin.php [hunk 1]
```diff
@@ -159,12 +158,12 @@ public function getUserName(string $accessToken): string
...
-        if (isset($user['localizedFirstName'])) {
-            $name = $user['localizedFirstName'];
+        if (isset($user['given_name'])) {
+            $name = $user['given_name'];
         }
 
-        if (isset($user['localizedLastName'])) {
-            $name = (empty($name)) ? $user['localizedLastName'] : $name . ' ' . $user['localizedLastName'];
+        if (isset($user['family_name'])) {
+            $name = (empty($name)) ? $user['family_name'] : $name . ' ' . $user['family_name'];
```
Comment (stnguyen90) [inline src/Appwrite/Auth/OAuth2/Linkedin.php:166]: Let's just use the name from LinkedIn:

```php
        if (isset($user['name'])) {
            return $user['name'];
        }
```

File: src/Appwrite/Auth/OAuth2/Linkedin.php [hunk 2]
```diff
@@ -144,9 +144,8 @@ public function getUserEmail(string $accessToken): string
      */
     public function isEmailVerified(string $accessToken): bool
     {
-        $email = $this->getUserEmail($accessToken);
-
-        return !empty($email);
+        $user = $this->getUser($accessToken);
+        return $user['email_verified'] ?? '';
```
Comment (stnguyen90) [inline src/Appwrite/Auth/OAuth2/Linkedin.php:148]: the function signature says this should return bool, but this line hear means it's possible for it to return a string.

```suggestion
        return $user['email_verified'] ?? false;
```

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Great PR! 🤯 We left some comments during the review, please check them out.

File: src/Appwrite/Auth/OAuth2/Linkedin.php (hunk 1 above)
Comment (btme0011) [reply src/Appwrite/Auth/OAuth2/Linkedin.php:166]: in some cases name might be not present and given_name or family_name might be present so should i add this case also with the other two cases? like if name is not present take info from given_name or family_name ?

File: src/Appwrite/Auth/OAuth2/Linkedin.php (hunk 2 above)
Comment (stnguyen90) [reply src/Appwrite/Auth/OAuth2/Linkedin.php:148]: Theoretically, an exception would be thrown elsewhere. For now, to keep things consistent, let's just return false.

File: src/Appwrite/Auth/OAuth2/Linkedin.php (hunk 1 above)
Comment (stnguyen90) [reply src/Appwrite/Auth/OAuth2/Linkedin.php:166]: Oh really? That's odd behavior...sure, go ahead and fallback to given and family name.

**Comment by venus-3** [comment]
@btme0011 

Can you please let me know when are you going to merge this PR, I am really waiting for this changes to be merged, I am experiencing the above issue while Integrating it with the Linkedin.

**Review by TorstenDittmann — APPROVED** [review]

**Review by stnguyen90 — APPROVED** [review]
I tested and was able to successfully log in:

image

**Pull request merged**